test/fzf.vader

```diff
@@ -147,6 +147,23 @@ Execute (fzf#wrap):
   let opts = fzf#wrap({})
   Assert opts.options =~ '^--color=fg:'
 
+Execute (fzf#shellesc with sh):
+  AssertEqual '""', fzf#shellesc('', 'sh')
+  AssertEqual '"\"\""', fzf#shellesc('""', 'sh')
+  AssertEqual '"foobar>"', fzf#shellesc('foobar>', 'sh')
+  AssertEqual '"\\""', fzf#shellesc('\"', 'sh')
```

Comment:
This PR becomes **Use fzf#shellescape for escaping shell arguments** to deprecate `s:shellesc` and the Window's version of `s:fzf_shellescape`. To be consistent with the built-in `shellescape`, I assume Windows-style env vars should  be escaped.